### PR TITLE
Update poverty and inequality datasets

### PR DESCRIPTION
**Pull request by paarriagadap**
- Modify LIS' Stata extracting code, to correct market income data and relative poverty.

- Update WID and LIS datasets

- Drop unused variables in LIS and WID explorers.

- A new `explorers` step is created, named `poverty_inequality`, which is a merge of both LIS and WID

### CODE REVIEW

File: etl/steps/data/garden/lis/2023-01-18/shared.py [hunk 1]
```diff
@@ -126,6 +126,20 @@
...
+    "share_bottom50": {
+        "title": "Share of the bottom 50%",
+        "description": "This is the income or consumption of the poorest 50% as a share of total income or consumption.",
+        "unit": "%",
+        "short_unit": "%",
+        "numDecimalPlaces": 1,
+    },
+    "share_middle40": {
+        "title": "Share of the middle 40%",
+        "description": "This is the income or consumption of the middle 40% as a share of total income or consumption.",
```
Comment (pabloarosado) [inline etl/steps/data/garden/lis/2023-01-18/shared.py:138]: I found this definition a bit odd (maybe it's a well-known metric, but to me it wasn't). I found a better explanation [here](https://wir2022.wid.world/category/glossary/):
"The middle 40% share is the share of income/wealth accruing to the middle 40% of the population, i.e. the population whose income/wealth lies above the median and below the top 10% income threshold."
Feel free to adapt your definition if you agree.

**Review by pabloarosado — APPROVED** [review]
Looks good, except for some minor changes and suggestions.

**Review by Marigold — APPROVED** [review]
Looks good, don't have anything more to add to Pablo's review.

File: etl/steps/data/garden/lis/2023-01-18/shared.py (hunk 1 above)
Comment (paarriagadap) [reply etl/steps/data/garden/lis/2023-01-18/shared.py:138]: You're right, I will add more details in that description, thanks!

File: etl/steps/data/explorers/poverty_inequality/2023-03-01/poverty_inequality.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+# define version of WID and LIS datasets
+WID_VERSION = "2023-01-27"
+LIS_VERSION = "2023-01-18"
+
+
+def run(dest_dir: str) -> None:
+
+    # Load WID explorer step
+    ds_wid = Dataset(DATA_DIR / "explorers" / "wid" / WID_VERSION / "world_inequality_database")
```
[the remark this answers is not in the document]
Comment (paarriagadap) [reply etl/steps/data/explorers/poverty_inequality/2023-03-01/poverty_inequality.py:25]: Ah, cool, I had the wrong idea about the function: I thought it only used previous steps when I was following the same name in the folders, and that's why I didn't use it. I will change it.

File: etl/steps/data/explorers/poverty_inequality/2023-03-01/poverty_inequality.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    tb_wid = ds_wid["world_inequality_database"]
+
+    # Load LIS explorer step
+    ds_lis = Dataset(DATA_DIR / "explorers" / "lis" / LIS_VERSION / "luxembourg_income_study")
+    tb_lis = ds_lis["luxembourg_income_study"]
+
+    # Merge both explorer datasets and assign a short name
+    tb_explorer = pd.merge(tb_wid, tb_lis, on=["country", "year"], how="outer")
```
[the remark this answers is not in the document]
Comment (paarriagadap) [reply etl/steps/data/explorers/poverty_inequality/2023-03-01/poverty_inequality.py:33]: Thanks will a `validate="one_to_one"` be enough? Both datasets have their indices with the integrity verified on `garden`

[the remark this answers is not in the document]
Comment (Marigold) [reply etl/steps/data/explorers/poverty_inequality/2023-03-01/poverty_inequality.py]: Whoa, I didn't even know such a parameter exists!

**Pull request merged**